src/mobu/models/jupyter.py

```diff
@@ -1,62 +1,161 @@
 """Models for configuring a Jupyter lab."""
 
-from enum import Enum
-from typing import Dict, Optional
+from enum import Enum, auto
+from typing import Dict, List, Optional
 
 from pydantic import BaseModel, Field, validator
 
-__all__ = ["JupyterConfig", "JupyterImage", "JupyterImageClass"]
+__all__ = [
+    "JupyterConfig",
+    "JupyterImageClass",
+    "ControllerImage",
+    "ControllerImages",
+]
 
 
-class JupyterImageClass(Enum):
-    """Possible ways of selecting an image."""
+def to_camel_case(string: str) -> str:
+    """Convert a string to camel case.
+
+    Originally written for use with Pydantic as an alias generator so that the
+    model can be initialized from camel-case input (such as Kubernetes
+    objects).
+
+    Parameters
+    ----------
+    string
+        Input string
+
+    Returns
+    -------
+    str
+        String converted to camel-case with the first character in lowercase.
+    """
+    components = string.split("_")
+    return components[0] + "".join(c.title() for c in components[1:])
+
+
+class CamelCaseModel(BaseModel):
+    """This is what we will use in place of BaseModel for the Spawner
+    Pydantic models.  Any configuration can be given in Helm-appropriate
+    camelCase, but internal Python methods and objects will all be snake_case.
+
+    This isn't actually all that useful here, but since these models are
+    copied from jupyterlabcontroller, which *does* make use of these features,
+    it's easier than messing with the models.
+    """
+
+    class Config:
+        """Pydantic configuration."""
+
+        alias_generator = to_camel_case
+        allow_population_by_field_name = True
```

Comment:
I don't think you end up needing any of this code since you use dashify instead.

Reply:
ControllerImages is a CamelCase model and I wanted to minimize the changes when I forklifted the model from jupyterlabcontroller.

That is a question I've got, actually: Ideally models shared between applications (e.g. as producer and consumer, which is what we're doing here) should go in some common library they both reference.  What's a good pattern for that?

Reply:
I've been pondering the same thing for Pydantic-based Avro for Roundtable Kafka producers and consumers. I was thinking about putting them in Safir?